src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -679,6 +680,41 @@ public void testExternalResourceFiltersWithNoExternalResource() throws Exception
         verify(checkerConfig, filePath, expected);
     }
 
+    /**
+     * This test is required for pitest coverage.
+     * Order of Check execution is not visible from outside
+     * as we order violations by line and column.
+     * This test is checking that Checks execution ordered by name and then by id.
```

Comment:
> Order of Check execution is not visible from outside

This is somewhat incorrect. This code came from random ordering of Javadoc checks. Multiple checks printed the same violation, but different executions picked which check was being printed as the source of the violation. This would cause differences in difference reports as we display who is the source of the violation.

The specific code from pitesting shows, we order **Checks** by class name, id, and then by hashcode. Those are the ordering of the execution of checks. Order of violations is defined elsewhere.

> ordered by name and then by id.

Show how this test ensures checks are ordered by id too? I only see somewhat reliance on check name ordering.

Reply:
The whole javadoc should just be:

`This test is checking that Checks execution ordered by name.`

Reply:
applied.